flux-attrs/src/lib.rs

```diff
@@ -76,10 +296,16 @@ fn create_dummy_fn(
     mangled_sig.ident = format_ident!("flux_extern_spec_{}", ident);
     let generic_call_args = transform_generic_params_to_call_args(sig.generics.params);
     let call_args = transform_params_to_call_args(sig.inputs);
-    let fn_path: syn::Path = if let Some(mod_path) = mod_path {
-        parse_quote_spanned!( ident.span() => #mod_path :: #ident)
-    } else {
-        parse_quote_spanned!( ident.span() => #ident )
+    let fn_path: syn::TypePath = match (mod_path, self_ty) {
```

Comment:
We can make a function 

```rust
fn qualified_path(mod_path: Option<syn::Path>, ty: syn::Type) -> SomeTypeHere {
    if let Some(mod_path) = mod_path {
        parse_quote_spanned! { self_ty_span => ( #mod_path :: #self_ty ) }
    } else {
        parse_quote_spanned! { self_ty_span => ( #self_ty ) }
    };
}
```
And then reuse it here and in `create_dummy_impl`

Reply:
Not sure about this one since the logic is slightly different (note how `self_ty` is wrapped in angle brackets). If you're convinced that is an acceptable substitute I'm happy to take it - or anything other than the enumeration I've done.

Reply:
I still think my comment applies because you add the angle brackets when used here but not when used in `create_dummy_impl`. Something like

```rust
let qualified_self_ty = qualified_path(mod_path, self_ty);
let fn_path = parse_quote_spanned!( ident.span() => < #qualified_self_ty > :: #ident ),
```

But I don't know much about `syn` to know if this works, or if it is a good idea for that matter. There may be some weirdness with spans.

Reply:
Your `qualified_path` function requires `self_ty` to be an `syn::Type`, but it's an `Option<syn::Type>`. I gave it some more thought and I couldn't reconcile the difference in the two cases. I'm going to take that as an indicator that they are disparate enough to justify not taking out a shared function.